### PR TITLE
Add URL slugs for events

### LINKED ISSUE #8: Add URL slugs to events
Currently event URLs use numeric IDs. These should be replaced with user-friendly slugs.

The edit form needs a new field, which can be auto-updated when editing the event name.

### PR DESCRIPTION
**Pull request by PurkkaKoodari**
Closes #8.

This PR adds an URL slug field to events and changes the user-visible URLs and APIs to use them. The admin side will still use event IDs, so that slugs can safely be edited in the event editor.

### CODE REVIEW

File: docs/migration.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+This file contains SQL scripts for migrating data from the Athene-created version of Ilmomasiina.
```
Comment (lakaniemi) [inline docs/migration.md:1]: I think we should also have some kind of automatic migration system too, as we will anyways run into the need of doing SQL migrations in the development process.

Comment (lakaniemi) [reply docs/migration.md:1]: Of course providing one-time migration script for people switching to this version is alright, but could it also be done using some tool that will also provide up/down migrations for future versions?

File: server/services/admin/slug/checkSlugAvailability.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import { Event } from '../../../models/event';
+
+export interface AdminCheckSlugResponse {
+  id: Event['id'] | null;
+  title: string | null;
+}
+
+/**
+ * @param slugOrId Event id if admin === true, slug if admin === false.
+ * @param admin Whether or not to return results for the admin view.
```
Comment (lakaniemi) [inline server/services/admin/slug/checkSlugAvailability.ts]: What are these related to? The function only has argument `slug`

File: server/services/admin/slug/index.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+/* eslint-disable class-methods-use-this */
+import { AdapterService } from '@feathersjs/adapter-commons';
+import { hooks as authHooks } from '@feathersjs/authentication';
+import { MethodNotAllowed } from '@feathersjs/errors';
+import { Id } from '@feathersjs/feathers';
+
+import { IlmoApplication } from '../../../defs';
+import checkSlugAvailability from './checkSlugAvailability';
+
+type AdminSlugServiceResponses = never;
```
Comment (lakaniemi) [inline server/services/admin/slug/index.ts]: Why is this typed never? :)

File: server/services/event/getEventDetails.ts [hunk 4]
```diff
@@ -118,21 +119,24 @@ export interface AdminEventGetResponse extends Pick<Event, typeof adminEventGetE
 
 export type EventGetResponseType<A extends boolean> = true extends A ? AdminEventGetResponse : EventGetResponse;
 
+/**
+ * @param slugOrId Event id if admin === true, slug if admin === false.
+ * @param admin Whether or not to return results for the admin view.
+ */
 export default async function getEventDetails<A extends boolean>(
-  id: number, admin: A,
+  slugOrId: string | number, admin: A,
```
Comment (lakaniemi) [inline server/services/event/getEventDetails.ts]: Not a fan of using single argument to mean two things in different scenarios. How about options-object pattern? 

```tsx
type GetEventDetailsOptions = {
  id?: string;
  slug?: string;
  admin: A;
}

// or if id or slug are required, maybe:

type GetEventDetailsOptions = {
  id: string;
  admin: A;
} | {
  slug: string;
  admin: A;
}

// Then function would have argument

fn(options: GetEventDetailsOptions)

// and called

fn({ slug: "dasdsa", admin: true })
```

File: server/services/event/getEventDetails.ts [hunk 5]
```diff
@@ -118,21 +119,24 @@ export interface AdminEventGetResponse extends Pick<Event, typeof adminEventGetE
 
 export type EventGetResponseType<A extends boolean> = true extends A ? AdminEventGetResponse : EventGetResponse;
 
+/**
+ * @param slugOrId Event id if admin === true, slug if admin === false.
+ * @param admin Whether or not to return results for the admin view.
+ */
 export default async function getEventDetails<A extends boolean>(
```
Comment (lakaniemi) [inline server/services/event/getEventDetails.ts]: Why is `A` even a type parameter? :D

File: docs/migration.md (hunk 1 above)
Comment (PurkkaKoodari) [reply docs/migration.md:1]: Yes and no. Sequelize has its own [migration system](https://sequelize.org/master/manual/migrations.html) but it doesn't support TypeScript and, as far as I can tell, relies somewhat on their CLI-generated models.

I'll investigate up/down migrations for new features when we get closer to deploying Ilmomasiina, but I don't think it's worth doing at this point when changes are happening rapidly.

File: server/services/admin/slug/checkSlugAvailability.ts (hunk 2 above)
Comment (PurkkaKoodari) [reply server/services/admin/slug/checkSlugAvailability.ts]: The Copypaste Strikes Back... fixed.

File: server/services/event/getEventDetails.ts (hunk 4 above)
Comment (PurkkaKoodari) [reply server/services/event/getEventDetails.ts]: Will definitely fix. The `getEventDetails` function has definitely seen too much evolution in its current form.

File: server/services/event/getEventDetails.ts (hunk 5 above)
Comment (PurkkaKoodari) [reply server/services/event/getEventDetails.ts]: This is so that we can return a different type based on whether we're getting stuff as an admin or user. Might be cleaner as a cast, but would lose typechecking.
